azavea/echo-locator
# Favorites support

## Overview

The existing site has a "favorites" function that allows users to keep a list of neighborhoods that they like. This implements the same functionality using the Django API.


### Checklist


### Demo

<img width="425" alt="Screen Shot 2022-05-20 at 13 20 49" src="https://user-images.githubusercontent.com/447977/169580177-32f4b748-2a3a-4aca-8ea7-2a247e0e5ce5.png">


### Notes

I opted not to remove the checks for the presence of `.favorites` mentioned in https://github.com/azavea/echo-locator/issues/515 -- they should now be superfluous but they're not hurting anything and could help the app be more resilient to failures in the future.


## Testing Instructions

 * Run `./scripts/update` and confirm migrations apply cleanly.
 * Log in to the admin site and confirm that you see a field in the `UserProfile` editor for favorites
 * If you've imported profiles from S3 to your local environment, use the admin to delete all of them
 * Run `AWS_PROFILE=echo-locator ./scripts/manage migrate_housing_seeker_data -b echolocator-staging-profiles122738-staging -p public` to import staging profiles to your local instance.
 * Click into a few of the newly imported files in the admin and confirm that some of them have favorites set.
 * Log in to the frontend
 * Use the ⭐  buttons next to a neighborhood card to add / remove some favorites. Edit other parts of your profile too. Reload the admin page for your profile and confirm that the favorites get persisted.
 * Log out and log back in and confirm that your favorites are saved.
 * Confirm that clicking "Saved" shows only your favorites.
 * Log out and browse anonymously and confirm that you can save favorites, and that they persist if you close the browser window and come back but are cleared if you logout.


Resolves #515
